cmd/example-guest-agent/main.go

```diff
@@ -0,0 +1,177 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	"github.com/mdlayher/vsock"
+	"google.golang.org/grpc"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/certificates/triple"
+	"kubevirt.io/kubevirt/pkg/certificates/triple/cert"
+	v1 "kubevirt.io/kubevirt/pkg/vsock/system/v1"
+)
+
+func main() {
+	log.InitializeLogging("example-guest-agent")
+
+	serverCert, err := setupCert()
+	if err != nil {
+		log.DefaultLogger().Reason(err).Errorf("count not generate the server TLS certificate")
+		os.Exit(1)
+	}
+
+	vsockConn, err := vsock.Listen(1234, &vsock.Config{})
```

Comment:
port 1?

Reply:
The agent is now used for the guest-agent and the current port settings should be consistent where needed. In general this is the guest-agent port and not the port of the system service (that is probably the confusion here).